ui/src/app/layout/topics/topicmetadatatable/topic-metadata-table.component.ts

```diff
@@ -41,21 +42,28 @@ export class TopicMetadataTableComponent implements OnInit {
 
     selectedSubscription: TopicSubscription;
 
+    selectedProducer: ApplicationInfo;
+
+    mappedProducers: ApplicationInfo[];
+
     constructor(
         private topicService: TopicsService,
         private environmentsService: EnvironmentsService,
         private translateService: TranslateService,
         private toasts: ToastService,
-        private modalService: NgbModal
+        private modalService: NgbModal,
+        private applicationsService: ApplicationsService
     ) {
 
     }
 
-    ngOnInit(): void {
+    async ngOnInit() {
+        this.mappedProducers = await Promise.all(this.topic.producers.map(producerId => this.applicationInfo(producerId)));
```

Comment:
Nope, this would block Angular rendering if backend (or my network connection) is slow. mappedProducers should be an observable.